cmd/gaia/init/validate_genesis.go

```diff
@@ -0,0 +1,48 @@
+package init
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/cmd/gaia/app"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/server"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/tendermint/tendermint/types"
+)
+
+// Validate genesis command takes
+func ValidateGenesisCmd(ctx *server.Context, cdc *codec.Codec) *cobra.Command {
+	return &cobra.Command{
+		Use:   "validate-genesis [file]",
+		Args:  cobra.RangeArgs(0, 1),
+		Short: "validates the genesis file at the default location or at the location passed as an arg",
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			var genesis string
+			var genDoc types.GenesisDoc
+			var genstate app.GenesisState
```

Comment:
These vars should be defined right before their first usage (aka `genesis` on line 31 etc.)

Reply:
👍